spec/messages_spec.rb

```diff
@@ -0,0 +1,83 @@
+require_relative "spec_helper"
+
+RSpec.describe Courier::Messages do
+  let(:client) { Courier::Client.new(AUTH_TOKEN_MOCK) }
+
+  context "list" do
+    it "succeeds without parameters" do
+      stub_request(:get, "https://api.courier.com/messages")
+        .with(
+          headers: TOKEN_AUTH_HEADERS
+        )
+        .to_return(body: "{\"paging\": {}, \"items\": []}", status: 200)
+      res = client.messages.list
+      expect(res).to eq({"paging" => {}, "items" => []})
+    end
+
+    it "succeeds with parameters" do
+      stub_request(:get, "https://api.courier.com/messages?cursor=12&event=34&recipient=56&notification=78&messageId=91&list=23")
+        .with(
+          headers: TOKEN_AUTH_HEADERS
+        )
+        .to_return(body: "{\"paging\": {}, \"items\": []}", status: 200)
+      res = client.messages.list(cursor: "12", event: "34", recipient: "56",
+                                 notification: "78", message_id: "91", list_id: "23")
+      expect(res).to eq({"paging" => {}, "items" => []})
+    end
+
+    it "fails with exception" do
+      stub_request(:get, "https://api.courier.com/messages")
+        .with(
+          headers: TOKEN_AUTH_HEADERS
+        ).to_return(body: "{\"message\": \"an error occurred\"}", status: 400)
+      expect { client.messages.list }.to raise_error(Courier::CourierAPIError)
```

Comment:
Nice-to-have, but you can also chain `.with_message` to `raise_error` in all of these and ensure that your error messaging is accurate on top of ensuring it raises the expected error. Far from essential but its a nice feature I leverage a lot myself with custom exceptions for library code.

Reply:
Thank you for all of the feedback!